Doc/tools/extensions/peg_highlight.py

```diff
@@ -0,0 +1,79 @@
+from pygments.lexer import RegexLexer, bygroups, include
+from pygments.token import Comment, Generic, Keyword, Name, Operator, Punctuation, Text
+
+from sphinx.highlighting import lexers
+
+
+class PEGLexer(RegexLexer):
+    """Pygments Lexer for PEG grammar (.gram) files
+
+    This lexer strips the following elements from the grammar:
+
+        - Meta-tags
+        - Variable assignments
+        - Actions
+        - Lookaheads
+        - Rule types
+        - Rule options
+        - Rules named `invalid_*` or `incorrect_*`
+    """
+
+    name = "PEG"
+    aliases = ["peg"]
+    filenames = ["*.gram"]
+    _name = r"([^\W\d]\w*)"
+    _text_ws = r"(\s*)"
+
+    tokens = {
+        "ws": [(r"\n", Text), (r"\s+", Text), (r"#.*$", Comment.Singleline),],
+        "lookaheads": [
+            (r"(&\w+\s?)", bygroups(None)),
+            (r"(&'.+'\s?)", bygroups(None)),
+            (r'(&".+"\s?)', bygroups(None)),
+            (r"(&\(.+\)\s?)", bygroups(None)),
+            (r"(!\w+\s?)", bygroups(None)),
+            (r"(!'.+'\s?)", bygroups(None)),
+            (r'(!".+"\s?)', bygroups(None)),
+            (r"(!\(.+\)\s?)", bygroups(None)),
+        ],
+        "metas": [
+            (r"(@\w+ '''(.|\n)+?''')", bygroups(None)),
+            (r"^(@.*)$", bygroups(None)),
+        ],
+        "actions": [(r"{(.|\n)+?}", bygroups(None)),],
+        "strings": [
+            (r"'\w+?'", Keyword),
+            (r'"\w+?"', Keyword),
```

Comment:
I feel that we'd better differentiate between hard and soft keywords. What about this?

```suggestion
            (r'"\w+?"', Keyword.Pseudo),
```

It would then look like this:

<img width="362" alt="Screenshot 2020-07-27 at 5 01 54 PM" src="https://user-images.githubusercontent.com/20306270/88557785-f53f6d80-d02a-11ea-84aa-ee7e65011227.png">

Although this is surely a discussion for somewhere else, I also feel that we should revisit the idea of having double quotes for soft keywords. I keep forgetting which is which.

Reply:
Let's put this discussion off until we actually *have* soft keywords (i.e. if/when PEP 622 is implemented).

Reply:
Yup, agreed.

What about the difference in highlighting? Do you want to go with this now or should we also care about this once we have soft keywords?

Reply:
I don't think we should use a different style for soft keywords.

Reply:
Ok.